Add omega scans to workflow

spxiwh opened the pull request:
This patch adds an omega scans executable to PyCBC and implements it within the PyCBC workflow. An example of this is here:

https://ldas-jobs.ligo.caltech.edu/~spxiwh/aLIGO/O2/analyses/O2_chunk19_HM/run3B/3._single_triggers/3.07_H1_loudest_noncoinc_time/

These new plots will appear in injection, single and coincident minifollowups.

This requires some minor changes to configuration files to add the new executable and options. Please advise on what file I should add those changes to?

tdent commented:
ok, on the colorbar: i suggest a default of log scaling and displaying one colorbar with log10 ticks, with a range up to the maximum over subplots, and I'd be OK with different subplots all using the same color scale.  (if you can persuade matplotlib to make a single colorbar beside 3 subplots ..)
I think this will work _because_ the log scaling should be able to show both quieter and louder glitches/signals at the same time.  And if there _is_ a louder glitch several seconds away from trigger merger time we'd want to know about it.

spxiwh commented:
@tdent Updated example to address your comment above:

https://ldas-jobs.ligo.caltech.edu/~spxiwh/H1-PLOT_QSCAN_2-1185217218-720000.png

What should the colorbar label say?

Adding the colorbar moves the x-axis label a little off-center. I think it's going to be a lot more effort to figure out how to resolve that than it's worth!

tdent commented:
'Normalized power' or 'Normalized energy' ? 
Off-center x label is fine IMO since most of the subplots arent centered at 0 anyway! 

BTW, would be nice to see the standard set of time windows around the 170817 trigger in each ifo as a test ..

spxiwh commented:
Hmmm:

https://ldas-jobs.ligo.caltech.edu/~spxiwh/H1-PLOT_QSCAN_2-1185217218-720000.png

https://ldas-jobs.ligo.caltech.edu/~spxiwh/L1-PLOT_QSCAN_2-1185217218-720000.png

tdent commented:
ah .. the qrange is totally wrong for BNS except for the lowest H1 plot. 
If you force q to be 50 or larger it will look 'good' but not sure what one can do about that in general.

tdent commented on bin/plotting/pycbc_plot_qscan:
```diff
@@ -0,0 +1,181 @@
+                         'default=4')
+parser.add_argument('--qtransform-qrange-upper', default=64, type=float,
+                    help='Upper limit of the range of q to consider, '
+                         'default=4')
```
presumably default=4 is a typo

spxiwh replied:
Fixed

titodalcanton commented on bin/plotting/pycbc_plot_qscan:
```diff
@@ -0,0 +1,181 @@
+                    version=pycbc.version.git_verbose_msg)
+parser.add_argument('--output-file', required=True, help='Output plot')
+parser.add_argument('--center-time', type=float,
+                    help='Center plot on the given GPS time')
```
Is `--center-time` not a required argument?

titodalcanton replied:
Nevermind, it makes sense after reading the other options.

spxiwh replied:
Added required=True here.

titodalcanton commented on bin/plotting/pycbc_plot_qscan:
```diff
@@ -0,0 +1,181 @@
+plt.tick_params(labelcolor='none', top='off', bottom='off', left='off',
+                right='off')
+plt.grid(False)
+plt.xlabel('Time from {} (s)'.format(opts.center_time))
```
I would use `{:.3f}` here to make sure we consistently get millisecond precision.

spxiwh replied:
Done

titodalcanton commented on bin/plotting/pycbc_plot_qscan:
```diff
@@ -0,0 +1,181 @@
+
+# https://stackoverflow.com/questions/13784201/matplotlib-2-subplots-1-colorbar
+cb = fig.colorbar(im, ax=axes.ravel().tolist())
+cb.set_label('What should this say?')
```
"Normalized energy", "Normalized signal energy", "Signal energy [au]"…

spxiwh replied:
Changed as suggested by Tom above

titodalcanton commented on bin/plotting/pycbc_plot_qscan:
```diff
@@ -0,0 +1,181 @@
+if opts.plot_caption is None:
+    # FIXME: Someone please improve!
+    opts.plot_caption = ("This shows the Q-transform as a function of time and "
+                        "frequency")
```
I would also store the center time in the default title and/or caption for reference, as written in the x label.

spxiwh replied:
Changed

titodalcanton commented:
As for the config file to add these options to, maybe the O3 development file?

https://github.com/ligo-cbc/pycbc-config/tree/master/O3/pipeline

spxiwh commented:
Indeed setting the lower qrange to 50 results in better plots:

https://ldas-jobs.ligo.caltech.edu/~spxiwh/H1-PLOT_QSCAN_3-1185217218-720000.png

https://ldas-jobs.ligo.caltech.edu/~spxiwh/L1-PLOT_QSCAN_3-1185217218-720000.png

spxiwh commented:
Okay, I think this is ready and I've addressed all comments now. I'll do one more quick test run and then look to push this with the requested config file changes.

spxiwh commented:
What's the benefit to a lower q-value than 50?

ahnitz commented:
I haven't quite followed how you are choosing q for each of the panes. Is that a separate choice? If  you are setting a blanket minimum at 50, you won't be able to effectively resolve any time-frequency evolution of BBH signals.

spxiwh commented:
Hi Alex,

I'm just following what timeseries.gtransform does. E.g. the "q" values are allowed to range between 4 and 64 (by default, which one can override on the command line, which gives the kwarg to timeseries.qtransform).

spxiwh commented:
https://github.com/ligo-cbc/pycbc-config/pull/27
